### ISSUE TITLE
clippy::integer_division_remainder_used fires on external macro code

### ISSUE THREAD

**Issue by rutgerclaes** [rust-lang/rust-clippy#17048]
### Summary

The restriction lint `clippy::integer_division_remainder_used` fires on `tokio::select!`.  The flagged `%` is not in my code: it comes from the macro's internal polling loop (`(start + i) % BRANCHES`), and I thus have no way to remove or rewrite it.

### Lint Name

integer_division_remainder_used

### Reproducer

I tried this code:

```rust
#![warn(clippy::integer_division_remainder_used)]
  #[tokio::main(flavor = "current_thread")]
  async fn main() {
      tokio::select! {
          () = tokio::time::sleep(std::time::Duration::from_millis(10)) => {}
          () = tokio::time::sleep(std::time::Duration::from_millis(20)) => {}
      }
  }
```

With 
```
tokio = { version = "1.52.3", features = ["macros", "rt", "time"] }
```

I saw this happen:

```
warning: use of `%` has been disallowed in this context
  --> src/main.rs:7:5
   |
 7 | /     tokio::select! {
 8 | |         () = tokio::time::sleep(Duration::from_millis(10)) => {}
 9 | |         () = tokio::time::sleep(Duration::from_millis(20)) => {}
10 | |     }
   | |_____^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#integer_division_remainder_used
note: the lint level is defined here
  --> src/main.rs:1:9
   |
 1 | #![warn(clippy::integer_division_remainder_used)]
```

I expected to see this happen:

The `%` is part of the select! macro, so as a user, I can not get rid of it.  The `select!` macro already has `#[allow(clippy::modulo_one)]`, but no `integer_division_remainder_used`: https://docs.rs/tokio/1.52.3/src/tokio/macros/select.rs.html#677

If I understand https://doc.rust-lang.org/nightly/clippy/development/macro_expansions.html correctly, clippy should not fire on external macro code.

> The general rule of thumb is that we should ignore code with macro expansions when working with Clippy because the code can be dynamic in ways that are difficult or impossible for us to foresee.

But I also undestand this is a very strict lint, so maybe it is intended to fire on generated code.

Based on the documentation, I feel this should be wrapped in `in_external_macro`.  If you agree, I can (try to) make a PR.

### Version

```text

```
